### PR TITLE
Modernize libs

### PR DESCRIPTION
**Pull request by StoneFrog**
- switches travis to workflows 
- tests against ruby 2.7+
- tests with new omniauth-oauth2
- added dependency on multi_json explicitly as omniauth-oauth2 versions (1.7, 1.8) failed as underlying gems removed their dependencies on it
- Minor specs fixes (leading slash removed here: https://gitlab.com/oauth-xx/oauth2/-/merge_requests/469, extra before action mirrors setup from other part of suite so just copied that)

### CODE REVIEW

File: spec/omniauth/strategies/bookingsync_spec.rb [hunk 1]
```diff
@@ -26,15 +26,19 @@
     end
 
     it "returns the correct authorization url" do
-      expect(strategy.client.options[:authorize_url]).to eq "/oauth/authorize"
+      expect(strategy.client.options[:authorize_url]).to eq "oauth/authorize"
```
Comment (Azdaroth) [inline spec/omniauth/strategies/bookingsync_spec.rb:29]: that's actually an important change, might be worth documenting. What caused it?

Comment (StoneFrog) [reply spec/omniauth/strategies/bookingsync_spec.rb:29]: @Azdaroth as mentioned in a PR description, it's originating from one of underlying gems:

**Review by Azdaroth — APPROVED** [review]
💪 🐉 Opa!

**Review by smilyalexey — APPROVED** [review]
LGTM

**Pull request merged**